### PR TITLE
Added api calls for hotels

### LINKED ISSUE #8: API call - hotels
GET 5-6 hotels
GET available hotels by startDate/endDate/city
GET hotel by id
GET all hotels

all of these DO NOT need auth

### PR DESCRIPTION
**Pull request by GoraJakub**

### CODE REVIEW

File: routes/hotels.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+const { Hotel } = require('../models/hotel')
+const router = express.Router()
+
+router.get('/', (req, res) => {
+  Hotel.find()
+    .limit(req.params.limit)
+    .exec()
+    .then((hotel) => {
+      if (!hotel) {
+        res.status(404).json({ message: 'No hotels in database' })
```
Comment (pdybowski) [inline routes/hotels.js]: If there are no hotels we should send just an empty array. Not throw an error

File: routes/hotels.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+const mongoose = require('mongoose')
+const express = require('express')
+const { Hotel } = require('../models/hotel')
+const router = express.Router()
+
+router.get('/', (req, res) => {
+  Hotel.find()
```
Comment (pdybowski) [inline routes/hotels.js]: Can you make it with await? It looks much better without using then. Please watch first link with JavaScript that should be in pd-przydatne-info.

File: routes/hotels.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    .limit(req.params.limit)
```
Comment (pdybowski) [inline routes/hotels.js]: We do not have limits in Params here

File: routes/hotels.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    })
+})
+
+router.get('/:limit', (req, res) => {
+  Hotel.find()
+    .limit(req.params.limit)
+    .exec()
+    .then((hotel) => {
+      if (!hotel) {
+        res.status(404).json({ message: 'No hotels in database' })
```
Comment (pdybowski) [inline routes/hotels.js]: Empty array

File: routes/hotels.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      }
+      res.status(200).json({ hotel: hotel })
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err })
+    })
+})
+
+router.get('/:limit', (req, res) => {
+  Hotel.find()
```
Comment (pdybowski) [inline routes/hotels.js]: Await

File: routes/hotels.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      }
+      res.status(200).json({ hotel: hotel })
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err })
+    })
+})
+
+router.get('/hotel/:hotelId', (req, res) => {
+  Hotel.findById(req.params.hotelId)
```
Comment (pdybowski) [inline routes/hotels.js]: Await

File: schemas/exampleSchema.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose')
+
+const exampleSchema = new mongoose.Schema({
```
Comment (pdybowski) [inline schemas/exampleSchema.js]: Remove this folder

**Review by pdybowski — COMMENTED** [review]
Zrob analogicznie dla Wszystskich

File: routes/hotels.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
+const mongoose = require('mongoose')
+const express = require('express')
+const { Hotel } = require('../models/hotel')
+const router = express.Router()
+
+router.get('/', async (req, res) => {
+  const hotel = await new Hotel.find()
```
Comment (pdybowski) [inline routes/hotels.js]: Hotels, and we don't need to have 'new' before hotel

File: routes/hotels.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+  res.status(200).send(hotel)
+})
+
+router.get('/:limit', async (req, res) => {
+  const limit = req.params.limit
+  const hotel = await new Hotel.find().limit(limit)
```
Comment (pdybowski) [inline routes/hotels.js]: Hotels, without new

File: routes/hotels.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  if (!hotel) {
+    return res.status(404).send('Hotel with provided ID not found')
+  }
+
+  res.status(200).send(hotel)
+})
+
+router.get('/city/:city', async (req, res) => {
+  const city = req.params.city
+  const hotel = await Hotel.find().where('localization.city', city)
```
Comment (pdybowski) [inline routes/hotels.js]: Hotels. Can you make it find({ localization.city: city })?

Please test it as well

File: routes/hotels.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+  if (!hotel) {
```
Comment (pdybowski) [inline routes/hotels.js]: Check what Hotel.find returns. If it's empty array then remove this error statement. Think that it throws empty array but check it

Comment (GoraJakub) [reply routes/hotels.js]: Yeah it returns empty array

**Pull request merged**